### PR TITLE
fix(ssh): preserve wildcard proxy endpoint fields in system SSH (#11746)

### LINKED ISSUE #11746: [P1 follow-up] ssh: wildcard ProxyCommand/ProxyJump still dials a bare alias on the system-SSH transport
Split out of #11707, which fixes the same bug on the **ssh2** transport only.

## What still reproduces

`shouldUseSystemSshTransport` (`src/main/ssh/ssh-connection.ts`) never consults `hasOpenSshHostBlockMatch`. For any config-backed target it returns `true` as soon as `resolved.proxyCommand` / `proxyJump` / `proxyUseFdpass` is set — and a wildcard `Host *` block supplies those for *every* alias, including one whose own `Host` block was renamed or removed. `buildSshArgs` then takes the `useConfigHost` branch (`src/main/ssh/system-ssh-args.ts:79-83`) and dials `target.configHost` verbatim.

Executed, not reasoned:

- target `{ source: 'ssh-config', configHost: 'prod', host: '10.0.0.5', port: 2222, username: 'deploy' }`
- resolved `{ hostname: 'prod', user: 'ops', port: 22, proxyCommand: 'nc -X connect …', forwardAgent: true, hostBlockMatch: false }` (only `Host *` supplies the proxy)

→ `shouldUseSystemSshTransport` = `true`, and `buildSshArgs` produces:

```
["-o","BatchMode=no","-T","-o","ControlMaster=auto","-o","ControlPath=…","-o","ControlPersist=300","-o","ServerAliveInterval=15","-o","ServerAliveCountMax=3","--","prod"]
```

No `-l deploy`, no `-p 2222`, no `-o Hostname=10.0.0.5`. Orca dials the bare alias as the wildcard's `ops` with the wildcard's `ForwardAgent`, discards the stored `10.0.0.5:2222/deploy` snapshot, and reports `connected` — the original P1 (wrong machine, agent identities offered to it), reached through the system-SSH consumer instead of ssh2.

## Why it was not fixed in #11707

The obvious one-liner — gating the config-backed branch of `shouldUseSystemSshTransport` on `hasOpenSshHostBlockMatch` — moves every wildcard-proxy user onto the ssh2 + `spawnProxyCommand` path wholesale, including users for whom that proxy is the only route to the network and users whose verdict is a false `false` (a real Host block that only restates wildcard values). That is a larger blast radius than the bug.

## Proposed direction

When `hostBlockMatch === false` **and** the stored endpoint differs from the alias, keep the system transport (so OpenSSH still applies the wildcard proxy) but pass the stored endpoint explicitly: `-o Hostname=<stored host>`, `-p <stored port>`, `-l <stored user>`.

Needs its own test matrix, at minimum:

- wildcard `ProxyCommand` with `%h` — the proxy now sees the stored host, not the alias (e.g. `cloudflared access ssh --hostname %h`)
- wildcard `ProxyJump`, and `ProxyUseFdpass`
- stored endpoint equal to the alias (must stay a no-op)
- false-`false` verdicts: a real `Host` block that only restates `Host *` values
- `getControlSocketPath` / ControlMaster socket identity
- the GitHub restricted-shell probe, relay deploy, and the sftp system fallbacks, which all go through `buildSshArgs`

Made with [Orca](https://github.com/stablyai/orca) 🐋

### PR DESCRIPTION
**Pull request by rodboev**
## Summary

- Preserve the saved host, port, and username when wildcard proxy settings keep a config-backed target on the system SSH transport after its concrete alias block no longer matches.
- Keep the alias as the OpenSSH config selector, so ProxyCommand, ProxyJump, and ProxyUseFdpass continue to run through the configured proxy.

Closes #11746.

## Screenshots

No visual change

## Testing

- [ ] `pnpm lint`
- [ ] `pnpm typecheck`
- [ ] `pnpm test`
- [ ] `pnpm build`
- [x] Added or updated high-quality tests covering the issue reproduction, proxy variants, concrete HostName precedence, default-port omission, and alias no-op behavior

Focused validation:

- [x] `pnpm exec vitest run --config config/vitest.config.ts src/main/ssh/system-ssh-resolved-endpoint.test.ts src/main/ssh/ssh-system-fallback.test.ts src/main/ssh/ssh-connection.test.ts`
- [x] `pnpm exec oxlint src/main/ssh/system-ssh-args.ts src/main/ssh/system-ssh-resolved-endpoint.test.ts`
- [x] `pnpm exec oxfmt --check src/main/ssh/system-ssh-args.ts src/main/ssh/system-ssh-resolved-endpoint.test.ts`
- [x] `pnpm run typecheck:node`
- [x] `pnpm run check:max-lines-ratchet`
- [x] `git diff --check`

## AI Review Report

The change is limited to system SSH argv construction. It keeps proxy selection, alias lookup, command placement, control-socket identity, and manual-target behavior unchanged, while passing the stored endpoint explicitly for the documented wildcard-proxy alias-default case. The argv remains discrete and platform-neutral across macOS, Linux, and Windows; no UI, Electron, provider, or release behavior changes.

## Security Audit

The fix prevents a stale alias/default result from routing a saved target and its configured SSH credentials to a different host. Host, port, and user remain separate argv values, and no value is interpolated into a shell command. Identity files, agent forwarding, proxy configuration, and control-master lifecycle remain under the existing OpenSSH configuration and process paths.

## Notes

The system-SSH slice is separate from https://github.com/stablyai/orca/pull/11707, which covers ssh2 transport resolution. Live remote OpenSSH and proxy expansion remain CI or maintainer-owned proof surfaces.

### CODE REVIEW

File: src/main/ssh/ssh-system-fallback.test.ts [hunk 1]
```diff
@@ -285,6 +285,73 @@ describe('spawnSystemSsh', () => {
...
+          'vpnuser',
+          '--',
+          'prod',
+          'echo ready'
+        ])
+      )
+      expect(args).not.toContain('22')
+      expect(args).not.toContain('deploy')
+    }
+  )
```
[the remark this answers is not in the document]
Comment (rodboev) [reply src/main/ssh/ssh-system-fallback.test.ts]: Confirmed, all three parameterized cases failed locally against the wrapped `exec /bin/sh -c ...` payload. They now pass `wrapCommand: false`, which keeps the assertion on the endpoint arguments this test is about. The cases also moved to `src/main/ssh/system-ssh-resolved-endpoint.test.ts`, since the additions pushed `ssh-system-fallback.test.ts` past the 800-line cap for test files.

**Comment by rodboev** [comment]
The current SSH argument builder preserves stored endpoint fields for wildcard matches, as implemented by merged PR https://github.com/stablyai/orca/pull/17946. This PR is superseded on current `main`.
